File: op-chain-ops/script/addresses/addresses.go
package addresses

import "github.com/ethereum/go-ethereum/common"

var (
	// DefaultSenderAddr is known as DEFAULT_SENDER = address(uint160(uint256(keccak256("foundry default caller"))))
	DefaultSenderAddr = common.HexToAddress("0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38")
	// DefaultScriptAddr is the address of the initial executing script, computed from:
	// cast compute-address --nonce 1 0x1804c8AB1F12E6bbf3894d4083f33e07309d1f38
	DefaultScriptAddr = common.HexToAddress("0x7FA9385bE102ac3EAc297483Dd6233D62b3e1496")
	// VMAddr is known as VM_ADDRESS = address(uint160(uint256(keccak256("hevm cheat code"))));
	VMAddr = common.HexToAddress("0x7109709ECfa91a80626fF3989D68f67F5b1DD12D")
	// ConsoleAddr is known as CONSOLE, "console.log" in ascii.
	// Utils like console.sol and console2.sol work by executing a staticcall to this address.
	ConsoleAddr = common.HexToAddress("0x000000000000000000636F6e736F6c652e6c6f67")
	// ScriptDeployer is used for temporary scripts address(uint160(uint256(keccak256("op-stack script deployer"))))
	ScriptDeployer = common.HexToAddress("0x76Ce131128F3616871f8CDA86d18fAB44E4d0D8B")
	// ForgeDeployer is used by some scripts as a default deployer address, e.g. makeAddr("deployer")
	ForgeDeployer = common.HexToAddress("0xaE0bDc4eEAC5E950B67C6819B118761CaAF61946")
)

File: rust/op-script-engine/src/addresses.rs
//! Well-known addresses, mirroring `op-chain-ops/script/addresses/addresses.go`
//! and `op-chain-ops/script/deterministic.go`.

use alloy_primitives::{Address, address};

/// hevm cheat-code precompile address (`keccak256("hevm cheat code")`).
pub const VM_ADDR: Address = address!("0x7109709ecfa91a80626ff3989d68f67f5b1dd12d");

/// The `console.log` sink ("console.log" in ascii).
pub const CONSOLE_ADDR: Address = address!("0x000000000000000000636f6e736f6c652e6c6f67");

/// Temporary script deployer (`keccak256("op-stack script deployer")`).
pub const SCRIPT_DEPLOYER: Address = address!("0x76ce131128f3616871f8cda86d18fab44e4d0d8b");

/// Default forge deployer address (`makeAddr("deployer")`).
pub const FORGE_DEPLOYER: Address = address!("0xae0bdc4eeac5e950b67c6819b118761caaf61946");

/// Foundry `DEFAULT_SENDER` (`keccak256("foundry default caller")`).
pub const DEFAULT_SENDER: Address = address!("0x1804c8ab1f12e6bbf3894d4083f33e07309d1f38");

/// Arachnid deterministic CREATE2 deployer used by forge broadcasts.
pub const CREATE2_DEPLOYER: Address = address!("0x4e59b44847b379578588920ca78fbf26c0b4956c");
